File: lib/feature/payment/domain/repository/payment_repository.dart
import 'package:dartz/dartz.dart';
import '../../../../shared/model/response_model.dart';
import '../model/response/payment_category.dart';
import '../model/response/payment_method.dart';

abstract class PaymentRepository {
  Future<Either<ResponseModel, List<PaymentCategory>>> getPaymentCategory();
  Future<Either<ResponseModel, List<PaymentResponse>>> getPaymentMethod({
    String? from,
    String? categoriesKey,
  });
  Future<Either<ResponseModel, PaymentResponse>> getDetailPaymentMethod();
}

File: src/features/payment/domain/repository/payment_repository.ts
import { Either } from "@/shared/utils/utility/either";
import { ResponseModel } from "@/shared/domain/model/response_model";
import { PaymentCategoryResponse } from "../model/response/payment_category_response";
import { PaymentMethodResponse } from "../model/response/payment_method_response";

export interface PaymentRepository {
  getPaymentCategory(token?: string): Promise<Either<ResponseModel, PaymentCategoryResponse[]>>;
  getPaymentMethod(
    params?: {
      paymentGatewayKey?: string;
      paymentGatewayId?: string;
      categoriesKey?: string;
    },
    token?: string,
  ): Promise<Either<ResponseModel, PaymentMethodResponse[]>>;
  getDetailPaymentMethod(key?: string, token?: string): Promise<Either<ResponseModel, PaymentMethodResponse>>;
}
